theano/sandbox/rng_mrg.py

```diff
@@ -804,101 +803,121 @@ def pretty_return(self, node_rstate, new_rstate, sample, size, nstreams):
-        high = cast(high, dtype=dtype)
+        low = tensor.cast(low, dtype=dtype)
+        high = tensor.cast(high, dtype=dtype)
 
-        low = undefined_grad(low)
-        high = undefined_grad(high)
+        # set random state
+        n_streams = nstreams or self.n_streams(size)
+        rstate_kwargs = dict(target=target) if target else {}
+        rstates = self.get_substream_rstates(n_streams, dtype=dtype)
+        node_rstate = theano.shared(rstates, **rstate_kwargs)
+        node_rstate.tag.is_rng = True
 
-        if isinstance(size, tuple):
-            msg = "size must be a tuple of int or a Theano variable"
-            assert all([isinstance(i, (np.integer, integer_types, Variable))
-                        for i in size]), msg
-            if any([isinstance(i, (np.integer, integer_types)) and i <= 0
-                    for i in size]):
-                raise ValueError(
-                    "The specified size contains a dimension with value <= 0",
-                    size)
+        # generate uniform samples
+        new_rstate = mrg_uniform.new(node_rstate, ndim, dtype, size)
+        u_samples = self.pretty_return(node_rstate, *new_rstate, size=size, nstreams=nstreams)
+        samples = u_samples * (high - low) + low
 
-        else:
-            if not (isinstance(size, Variable) and size.ndim == 1):
-                raise TypeError("size must be a tuple of int or a Theano "
-                                "Variable with 1 dimension, got " + str(size) +
-                                " of type " + str(type(size)))
-        orig_nstreams = nstreams
-        if nstreams is None:
-            nstreams = self.n_streams(size)
-        rstates = self.get_substream_rstates(nstreams, dtype)
-
-        d = {}
-        if 'target' in kwargs:
-            d = dict(target=kwargs.pop('target'))
-        if len(kwargs) > 0:
-            raise TypeError("uniform() got unexpected keyword arguements %s" % (str(kwargs.keys())))
-        node_rstate = shared(rstates, **d)
-        u = self.pretty_return(node_rstate,
-                               *mrg_uniform.new(node_rstate,
-                                                ndim, dtype, size),
-                               size=size, nstreams=orig_nstreams)
-        # Add a reference to distinguish from other shared variables
-        node_rstate.tag.is_rng = True
-        r = u * (high - low) + low
+        if u_samples.type.broadcastable != samples.type.broadcastable:
+            raise NotImplementedError("Increase the size to match the broadcasting pattern "
+                                      "of `low` and `high` arguments")
 
-        if u.type.broadcastable != r.type.broadcastable:
-            raise NotImplementedError(
-                'Increase the size to match the broadcasting pattern of '
-                '`low` and `high` arguments')
+        return samples
 
-        assert r.dtype == dtype
-        return r
+    def binomial(self, size, n=1, p=0.5,
```

Comment:
There are currently no tests for `binomial()` when `n > 1`, in `theano/sandbox/tests/test_rng_mrg.py`. Could you add a test for that case ?